robyn/openapi.py

```diff
@@ -0,0 +1,275 @@
+
+        modified_endpoint, path_obj = self.get_path_obj(endpoint, openapi_summary, openapi_tags, query_params)
+
+        if modified_endpoint not in self.openapi_spec["paths"]:
+            self.openapi_spec["paths"][modified_endpoint] = {}
+        self.openapi_spec["paths"][modified_endpoint][route_type] = path_obj
+
+    def add_subrouter_paths(self, subrouter_openapi):
+        """
+        Adds the subrouter paths to main router's openapi specs
+
+        :param subrouter_openapi: the OpenAPI object of the current subrouter
+        """
+        paths = subrouter_openapi.openapi_spec["paths"]
+
+        for path in paths:
+            self.openapi_spec["paths"][path] = paths[path]
+
+    def get_path_obj(self, endpoint: str, summary: str, tags: list, query_params: TypedDict = None) -> (str, dict):
+        """
+        Get the "path" openapi object according to spec
+
+        :param endpoint: the endpoint to be added
+        :param summary: short summary of the endpoint (to be fetched from the endpoint defenition by default)
+        :param tags: tags -- for grouping of endpoints
+        :param query_params: query params for the function
+
+        :return: a tuple containing the endpoint with path params wrapped in braces and the "path" openapi object
+        according to spec
+        """
+        # robyn has paths like /:url/:etc whereas openapi requires path like /{url}/{path}
+        # this function is used for converting path params to the required form
+        # initialized with endpoint for handling endpoints without path params
+        endpoint_with_path_params_wrapped_in_braces = endpoint
+
+        endpoint_path_params_split = endpoint.split(":")
+        openapi_parameter_object = []
+
+        if len(endpoint_path_params_split) > 1:
+            endpoint_without_path_params = endpoint_path_params_split[0]
+
+            endpoint_with_path_params_wrapped_in_braces = (
+                endpoint_without_path_params[:-1] if endpoint_without_path_params.endswith("/") else endpoint_without_path_params
+            )
+
+            for path_param in endpoint_path_params_split[1:]:
+                path_param_name = path_param[:-1] if path_param.endswith("/") else path_param
+
+                openapi_parameter_object.append(
+                    {
+                        "name": path_param_name,
+                        "in": "path",
+                        "required": True,
+                        "schema": {"type": "string"},
+                    }
+                )
+                endpoint_with_path_params_wrapped_in_braces += "/{" + path_param_name + "}"
+
+        if query_params:
+            for query_param in query_params.__annotations__:
+                # ugly hack!
+                # returns "<class 'int'>" -- this line strips out the type a.k.a int from it
+                query_param_type = re.findall("'[a-z]+'", str(query_params.__annotations__[query_param]))[0].replace("'", "")
```

Comment:
still need to have a look at this

Reply:
hey @sansyrox I can't figure out a way to do it, can you give some help & point me in the right direction?!